code42_connector.py

```diff
@@ -0,0 +1,156 @@
+# Python 3 Compatibility imports
+from __future__ import print_function, unicode_literals
+
+# Phantom App imports
+import phantom.app as phantom
+from phantom.base_connector import BaseConnector
+from phantom.action_result import ActionResult
+
+# Usage of the consts file is recommended
+# from code42_consts import *
+import requests
+import json
+
+import py42.sdk
+import py42.util as util
+
+
+class RetVal(tuple):
+
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class Code42Connector(BaseConnector):
+
+    def __init__(self):
+        super(Code42Connector, self).__init__()
+
+        self._state = None
+        self._cloud_instance = None
+        self._username = None
+        self._password = None
+        self._client = None
+
+    def _create_client(self, action_result):
+        try:
+            self._client = py42.sdk.from_local_account(self._cloud_instance, self._username, self._password)
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, f"Could not create py42 client: {e}")
+
+        return phantom.APP_SUCCESS
+
+    def _handle_test_connectivity(self, param):
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        self.save_progress("Connecting to endpoint")
+
+        if not self._create_client(action_result):
+            return action_result.get_status()
+
+        response = self._client.users.get_current()
+        util.print_response(response)
+
+        self.save_progress("Test Connectivity Passed")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def handle_action(self, param):
+        ret_val = phantom.APP_SUCCESS
+
+        # Get the action that we are supposed to execute for this App Run
+        action_id = self.get_action_identifier()
+
+        self.debug_print("action_id", self.get_action_identifier())
+
+        if action_id == 'test_connectivity':
+            ret_val = self._handle_test_connectivity(param)
+
+        return ret_val
+
+    def initialize(self):
```

Comment:
can we order these functions in a way that makes it easier to understands what's going on if you read this from top to bottom:

```
__init__
initialize
handle_action
_create_client
_handle_test_connectivity
# future action handlers go here
```

for example, when `_create_client` is right under `__init__`, it kind of looks like the client could get created with a `_cloud_instance` of `None`, even though that woudn't actually happen